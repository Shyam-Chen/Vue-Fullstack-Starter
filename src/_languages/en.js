export default {
  /** @name practical-examples */
  practicalExamples: 'Practical Examples',

  overview: 'Overview',
  examples: 'Examples',

  helloWorld: 'Hello World',

  guide: 'Guide',
  templateRendering: 'Template Rendering',
  components: 'Components',
  props: 'Props',
  slots: 'Slots',
  refs: 'Refs',
  customEvents: 'Custom Events',
  extends: 'Extends',
  dynamic: 'Dynamic',
  recursion: 'Recursion',
  lifecycleHooks: 'Lifecycle Hooks',
  transitions: 'Transitions',
  mixins: 'Mixins',
  directives: 'Directives',
  dependencyInjection: 'Dependency Injection',
  filters: 'Filters',
  routing: 'Routing',
  dynamicMatching: 'Dynamic Matching',
  nestedRoutes: 'Nested Routes',
  guards: 'Guards',
  stateManagement: 'State Management',

  crudOperations: 'CRUD Operations',
  basic: 'Basic',
  rest: 'REST',
  graphql: 'GraphQL',

  controls: 'Controls',
  textFields: 'Text Fields',
  selects: 'Selects',
  selectionControls: 'Selection Controls',
  pickers: 'Pickers',
  autocomplete: 'Autocomplete',
  stepper: 'Stepper',
  tabs: 'Tabs',
  treeView: 'Tree View',
  calendar: 'Calendar',
  markdownEditor: 'Markdown Editor',
  richTextEditor: 'Rich Text Editor',

  dataTable: 'Data Table',
  displayRows: 'Display Rows',
  inlineEdit: 'Inline Edit',
  paginator: 'Paginator',
  iterator: 'Iterator',
  exportReport: 'Export Report',

  internationalization: 'Internationalization',

  dataVisualization: 'Data Visualization',

  authorization: 'Authorization',

  payment: 'Payment',

  messages: 'Messages',
  chatRoom: 'Chat Room',
  email: 'Email',
  cloudMessaging: 'Cloud Messaging',

  machineLearning: 'Machine Learning',

  /** @name additional-resources */
  additionalResources: 'Additional Resources',

  universalApps: 'Universal Apps',
  mobileApps: 'Mobile Apps',
  desktopApps: 'Desktop Apps',
};
