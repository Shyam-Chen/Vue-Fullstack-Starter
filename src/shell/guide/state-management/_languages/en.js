export default {
  increment: 'Increment',
  decrement: 'Decrement',
  async: 'Async',
  ifOdd: 'If Odd',
  ifEven: 'If Even',
};
