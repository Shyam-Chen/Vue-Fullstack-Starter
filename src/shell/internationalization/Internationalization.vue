<template>
  <v-container class="shell-container" fluid>
    <v-layout column>
      <div class="headline">Internationalization</div>

      <v-layout row>
        <v-btn @click="setLanguage('en')">English</v-btn>
        <v-btn @click="setLanguage('zh')">中文</v-btn>
        <v-btn @click="setLanguage('ja')">日本語</v-btn>
      </v-layout>

      <v-layout row>
        <p class="pl-2 pt-4">
          HTML formatting:
          <b class="primary--text">{{ $t('bubble') }}</b>
        </p>
      </v-layout>

      <v-layout row>
        <p class="pl-2">
          Named formatting:
          <b class="primary--text">{{ $t('hello', { msg: 'Vue' }) }}</b>
        </p>
      </v-layout>

      <v-layout row>
        <p class="pl-2">
          List formatting:
          <b class="primary--text">{{ $t('thinkIn', ['Angular', 'React', 'Vue']) }}</b>
        </p>
      </v-layout>

      <v-layout row>
        <p class="pl-2">
          Pluralization:
          <b class="primary--text">{{ $tc('http', 2) }}://google.com</b>
        </p>
      </v-layout>

      <v-layout row>
        <p class="pl-2">
          Pluralizing i18n:
          <b class="primary--text">{{ $tc('framework', 0) }}</b
          >, <b class="primary--text">{{ $tc('framework', 1) }}</b
          >,
          <b class="primary--text">{{ $tc('framework', 3, { num: 3 }) }}</b>
        </p>
      </v-layout>
    </v-layout>
  </v-container>
</template>

<script>
import { mapActions } from 'vuex';

export default {
  methods: {
    ...mapActions(['setLanguage']),
  },
};
</script>
