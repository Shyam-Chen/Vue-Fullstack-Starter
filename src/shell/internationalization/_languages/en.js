export default {
  bubble: 'Bubble',
  hello: 'Hello, {msg}!',
  thinkIn: 'Think in {2}',
  http: 'http | https',
  framework: 'No framework | One framework | {num} frameworks',
};
