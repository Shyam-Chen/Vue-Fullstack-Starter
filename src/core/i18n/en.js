import app from '~/_languages/en';
import home from '~/home/_languages/en';
import guideStateManagement from '~/shell/guide/state-management/_languages/en';
import internationalization from '~/shell/internationalization/_languages/en';

export default {
  ...app,
  home,
  ...guideStateManagement,
  ...internationalization,
};
