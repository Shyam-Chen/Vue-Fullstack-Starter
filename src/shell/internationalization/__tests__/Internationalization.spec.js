import { createLocalVue, shallowMount } from '@vue/test-utils';
import I18n from 'vue-i18n';

import Internationalization from '../Internationalization';
import en from '../_languages/en';

const localVue = createLocalVue();

localVue.use(I18n);

describe('Internationalization', () => {
  let [wrapper, i18n] = [];

  beforeEach(() => {
    i18n = new I18n({
      locale: 'en',
      messages: { en },
    });

    wrapper = shallowMount(Internationalization, { localVue, i18n });
  });

  it('should render an initial component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
