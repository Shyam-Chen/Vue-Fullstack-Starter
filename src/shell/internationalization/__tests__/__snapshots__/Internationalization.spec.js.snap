// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Internationalization should render an initial component 1`] = `
<v-container-stub tag="div" fluid="true" class="shell-container">
  <v-layout-stub tag="div" column="">
    <div class="headline">Internationalization</div>
    <v-layout-stub tag="div" row="">
      <v-btn-stub tag="button" activeclass="" type="button">English</v-btn-stub>
      <v-btn-stub tag="button" activeclass="" type="button">中文</v-btn-stub>
      <v-btn-stub tag="button" activeclass="" type="button">日本語</v-btn-stub>
    </v-layout-stub>
    <v-layout-stub tag="div" row="">
      <p class="pl-2 pt-4">
        HTML formatting:
        <b class="primary--text">Bubble</b></p>
    </v-layout-stub>
    <v-layout-stub tag="div" row="">
      <p class="pl-2">
        Named formatting:
        <b class="primary--text">Hello, Vue!</b></p>
    </v-layout-stub>
    <v-layout-stub tag="div" row="">
      <p class="pl-2">
        List formatting:
        <b class="primary--text">Think in Vue</b></p>
    </v-layout-stub>
    <v-layout-stub tag="div" row="">
      <p class="pl-2">
        Pluralization:
        <b class="primary--text">https://google.com</b></p>
    </v-layout-stub>
    <v-layout-stub tag="div" row="">
      <p class="pl-2">
        Pluralizing i18n:
        <b class="primary--text">No framework</b>,
        <b class="primary--text">One framework</b>,
        <b class="primary--text">3 frameworks</b></p>
    </v-layout-stub>
  </v-layout-stub>
</v-container-stub>
`;
